routes/users.js

```diff
@@ -272,12 +316,60 @@ router.get('/terms', async (req, res) => {
 	}
 });
 
+// @desc    Render login.html
+// @route   GET '/users/verify'
+// @access  Public
+// @tested 	Not yet
+router.get('/verify/:hash', async (req, res) => {
+
+	try {
+		const user = await User.findOne({
+			verificationHash: req.params.hash
+		});
+
+		if (user) {
+			if (user.emailVerfied) {
+
+				return res.status(200).render('login', {
+					user: res.locals.user,
+					successNotification: {msg: "Your email is already verified, you can login now!"},
+					errorNotification: null
+				});
+			}
+			user.emailVerfied = true;
+			user.save();
+
+			return res.status(200).render('login', {
+				user: res.locals.user,
+				successNotification: {msg: "Verification successful, you can login now!"},
+				errorNotification: null
+			});
+		} else {
+
+			return res.status(400).render('login', {
+				user: res.locals.user,
+				successNotification: null,
+				errorNotification:  {msg: "Verfication failed"}
```

Comment:
Don't forget the i in Verification for this one and one on line 361 :) Esp since it's user facing.